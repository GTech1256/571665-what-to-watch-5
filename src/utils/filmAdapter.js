/* eslint-disable camelcase */

export const adaptFilmToServer = ({
  id,
  name,
  posterImage,
  previewImage,
  backgroundImage,
  backgroundColor,
  videoLink,
  previewVideoLink,
  description,
  rating,
  scoresCount,
  director,
  starring,
  runTime,
  genre,
  released,
  isFavorite
}) => ({
  id,
  name,
  poster_image: posterImage,
  preview_image: previewImage,
  background_image: backgroundImage,
  background_color: backgroundColor,
  description,
  rating,
  scores_count: scoresCount,
  director,
  starring,
  run_time: runTime,
  genre,
  released,
  is_favorite: isFavorite,
  video_link: videoLink,
  preview_video_link: previewVideoLink
});


export const adaptFilmFromServer = ({
  id,
  name,
  poster_image,
  preview_image,
  background_image,
  background_color,
  description,
  rating,
  scores_count,
  director,
  starring,
  run_time,
  genre,
  released,
  is_favorite,
  video_link,
  preview_video_link
}) => ({
  id,
  name,
  posterImage: poster_image,
  previewImage: preview_image,
  backgroundImage: background_image,
  backgroundColor: background_color,
  videoLink: video_link,
  previewVideoLink: preview_video_link,
  description,
  rating,
  scoresCount: scores_count,
  director,
  starring,
  runTime: run_time,
  genre,
  released,
  isFavorite: is_favorite
});

