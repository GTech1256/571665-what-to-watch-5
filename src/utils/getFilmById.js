import films from "../mocks/films";

export const getFilmById = (filmId) => films.find(({id}) => filmId === `${id}`);
