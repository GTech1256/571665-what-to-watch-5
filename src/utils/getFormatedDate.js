const DATE_TIME_FORMAT_LOCALE = `en-US`;
const DATE_TIME_FORMAT_OPTION = {
  month: `long`,
  day: `numeric`,
  year: `numeric`
};

export const getFormatedDate = (value) => new Intl.DateTimeFormat(
    DATE_TIME_FORMAT_LOCALE,
    DATE_TIME_FORMAT_OPTION
)
  .format(
      new Date(value)
  );
