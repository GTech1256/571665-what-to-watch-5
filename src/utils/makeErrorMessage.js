export const makeErrorMessage = (error) => `${error.message}${error.response ? `: ${error.response.data.error}` : ``}`;
