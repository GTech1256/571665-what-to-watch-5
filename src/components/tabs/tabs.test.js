import React from "react";
import renderer from "react-test-renderer";
import {noop} from "../../utils/noop";
import Tab from "../tab/tab";
import Tabs from "./tabs";

const ComponentMock = () => <p>ComponentMock</p>;
const ACTIVE_ITEM_INDEX = 0;

it(`Tabs is rendered correctly`, () => {
  const tree = renderer
    .create(
        <Tabs
          activeItem={ACTIVE_ITEM_INDEX}
          onActiveItemChange={noop}
        >
          <Tab
            title="ComponentMock"
          >
            <ComponentMock />
          </Tab>
          <Tab
            title="test"
          >
            <p>test</p>
          </Tab>
        </Tabs>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
