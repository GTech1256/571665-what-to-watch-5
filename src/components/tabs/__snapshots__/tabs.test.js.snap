// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs is rendered correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          ComponentMock
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          onClick={[Function]}
        >
          test
        </a>
      </li>
    </ul>
  </nav>
  <p>
    ComponentMock
  </p>
</div>
`;
